crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -1115,6 +1115,21 @@ where
                 // at the end to match the runtime evaluation of parameter defaults
                 // and return-type annotations.
                 let (symbol, _) = self.add_symbol(name.id.clone());
+
+                // Record a use of the function name in the scope that it is defined in, so that it
+                // can be used to find previously defined functions with the same name. This is
+                // used to collect all the overloaded definitions of a function. This needs to be
+                // done on the `Identifier` node as oppose to `ExprName` because that's what the
+                // AST uses.
+                //
+                // This could possibly be done only if there are decorators to avoid the extra
+                // work for non-overloaded functions, but that wouldn't work because the
+                // implementation function is not decorated with `@overload`.
```

Comment:
Seems confusing to say "this could be done" and then explain why it couldn't; I'd probably just remove this.